File: src/main/java/net/gamershub/kitpvp/items/CustomItemHandler.java
package net.gamershub.kitpvp.items;

import net.gamershub.kitpvp.PDCUtils;
import net.gamershub.kitpvp.Utils;
import net.gamershub.kitpvp.items.impl.artilleryman.Jetpack;
import net.gamershub.kitpvp.items.impl.artilleryman.RocketLauncher;
import net.gamershub.kitpvp.items.impl.assassin.AssassinSword;
import net.gamershub.kitpvp.items.impl.creeper.CreeperLeggings;
import net.gamershub.kitpvp.items.impl.creeper.FireballSword;
import net.gamershub.kitpvp.items.impl.devil.DevilsSword;
import net.gamershub.kitpvp.items.impl.enderman.EnderSword;
import net.gamershub.kitpvp.items.impl.engineer.TrapWand;
import net.gamershub.kitpvp.items.impl.engineer.TurretItem;
import net.gamershub.kitpvp.items.impl.magician.MagicWand;
import net.gamershub.kitpvp.items.impl.poseidon.PoseidonsTrident;
import net.gamershub.kitpvp.items.impl.provoker.NukeItem;
import net.gamershub.kitpvp.items.impl.provoker.SpookSword;
import net.gamershub.kitpvp.items.impl.tank.TankAxe;
import net.gamershub.kitpvp.items.impl.tank.TankBoots;
import net.gamershub.kitpvp.items.impl.vampire.VampireSword;
import net.gamershub.kitpvp.items.impl.zeus.LightningWand;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.HashMap;
import java.util.Map;

public class CustomItemHandler {
    public Map<String, CustomItem> ID_MAP = new HashMap<>();

    public static CustomItem FIREBALL_SWORD;
    public static CustomItem LIGHTNING_WAND;
    public static CustomItem MAGIC_WAND;
    public static CustomItem VAMPIRE_SWORD;
    public static CustomItem NUKE;
    public static CustomItem SPOOK_SWORD;
    public static CustomItem TRAP_WAND;
    public static CustomItem CREEPER_LEGGINGS;
    public static CustomItem ENDER_SWORD;
    public static CustomItem POSEIDONS_TRIDENT;
    public static CustomItem DEVILS_SWORD;
    public static CustomItem TANK_BOOTS;
    public static CustomItem TANK_AXE;
    public static CustomItem JETPACK;
    public static CustomItem ASSASSIN_SWORD;
    public static CustomItem ROCKET_LAUNCHER;
    public static CustomItem TURRET;

    public CustomItemHandler() {
        FIREBALL_SWORD = registerItem(new FireballSword());
        LIGHTNING_WAND = registerItem(new LightningWand());
        MAGIC_WAND = registerItem(new MagicWand());
        VAMPIRE_SWORD = registerItem(new VampireSword());
        NUKE = registerItem(new NukeItem());
        SPOOK_SWORD = registerItem(new SpookSword());
        TRAP_WAND = registerItem(new TrapWand());
        CREEPER_LEGGINGS = registerItem(new CreeperLeggings());
        ENDER_SWORD = registerItem(new EnderSword());
        POSEIDONS_TRIDENT = registerItem(new PoseidonsTrident());
        DEVILS_SWORD = registerItem(new DevilsSword());
        TANK_BOOTS = registerItem(new TankBoots());
        TANK_AXE = registerItem(new TankAxe());
        JETPACK = registerItem(new Jetpack());
        ASSASSIN_SWORD = registerItem(new AssassinSword());
        ROCKET_LAUNCHER = registerItem(new RocketLauncher());
        TURRET = registerItem(new TurretItem());
    }

    private CustomItem registerItem(CustomItem item) {
        Utils.registerEvents(item);
        ID_MAP.put(item.id, item);
        return item;
    }

    @Nullable
    public static String getCustomItemId(@NotNull ItemStack itemStack) {
        return PDCUtils.getId(itemStack.getItemMeta());
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/items/CustomItemHandler.kt
package io.github.chaosdave34.kitpvp.items

import io.github.chaosdave34.kitpvp.KitPvp
import io.github.chaosdave34.kitpvp.Utils
import io.github.chaosdave34.kitpvp.items.impl.*
import org.bukkit.NamespacedKey
import org.bukkit.inventory.ItemStack
import org.bukkit.persistence.PersistentDataType

class CustomItemHandler {
    val customItems: MutableMap<String, CustomItem> = mutableMapOf()

    companion object {
        lateinit var MAGIC_WAND: CustomItem
        lateinit var VAMPIRE_SWORD: CustomItem
        lateinit var CREEPER_LEGGINGS: CustomItem
        lateinit var ENDER_SWORD: CustomItem
        lateinit var DEVILS_SWORD: CustomItem
        lateinit var TANK_BOOTS: CustomItem
        lateinit var JETPACK: CustomItem
        lateinit var ASSASSIN_SWORD: CustomItem
        lateinit var ROCKET_LAUNCHER: CustomItem
        lateinit var LONG_BOW: CustomItem
        lateinit var ENGINEERS_SWORD: CustomItem

        fun ItemStack.getCustomItemId(): String? = this.persistentDataContainer.get(NamespacedKey(KitPvp.INSTANCE, "id"), PersistentDataType.STRING)
    }

    init {
        MAGIC_WAND = registerItem(MagicWand())
        VAMPIRE_SWORD = registerItem(VampireSword())
        CREEPER_LEGGINGS = registerItem(CreeperLeggings())
        ENDER_SWORD = registerItem(EnderSword())
        DEVILS_SWORD = registerItem(DevilsSword())
        TANK_BOOTS = registerItem(TankBoots())
        JETPACK = registerItem(Jetpack())
        ASSASSIN_SWORD = registerItem(AssassinSword())
        ROCKET_LAUNCHER = registerItem(RocketLauncher())
        LONG_BOW = registerItem(LongBow())
        ENGINEERS_SWORD = registerItem(EngineersSword())
    }

    private fun registerItem(item: CustomItem): CustomItem {
        Utils.registerEvents(item)
        customItems[item.id] = item
        return item
    }
}